### PR TITLE
feat: support notifications on rollout events using notifications-engine

### PR DESCRIPTION
**Pull request by alexmt**
Closes https://github.com/argoproj/argo-rollouts/issues/369

Signed-off-by: Alexander Matyushentsev <AMatyushentsev@gmail.com>

* [x] Code changes
* [x] Unit tests
* [x] Docs

PR adds notifications support to the Rollout. Following changes are implemented:

* [Event Recorder](https://github.com/alexmt/argo-rollouts/blob/369-notifications-custom-triggers/utils/record/record.go) translates the specified event reason to the trigger and sends a notification if the trigger has configured template
* [Catalog](https://github.com/alexmt/argo-rollouts/tree/369-notifications-custom-triggers/manifests/notifications) of notifications for several built-in triggers . Can be optionally installed by user
* Custom Triggers support: https://github.com/alexmt/argo-rollouts/blob/369-notifications-custom-triggers/controller/controller.go#L153
* Documentation that explains how to start using notifications: https://github.com/alexmt/argo-rollouts/blob/369-notifications-custom-triggers/docs/features/notifications.md

### CODE REVIEW

File: docs/features/notifications.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: argo-rollouts-notification-secret
+stringData:
+  slack-token: <my-slack-token>
+```
+
+Learn more about supported services and configuration settings in services [documentation](../generated/notification-services/overview.md).
```
Comment (khhirani) [inline docs/features/notifications.md:46]: This link is broken

**Review by jessesuen — APPROVED** [review]

File: docs/features/notifications.md (hunk 1 above)
Comment (alexmt) [reply docs/features/notifications.md:46]: link is referencing file is generated directory. So these files are not committed to git.

File: docs/features/notifications.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+# Notifications
```
[the remark this answers is not in the document]
Comment (alexmt) [reply docs/features/notifications.md:1]: done

[the remark this answers is not in the document]
Comment (jtai-omniex) [reply docs/features/notifications.md:1]: has v1.1 been released yet? i see that the release tags only go up to v1.0.2

**Pull request merged**